capa/rules/__init__.py

```diff
@@ -194,6 +196,78 @@ def __repr__(self):
         return str(self)
 
 
+class ComType(Enum):
+    CLASS = "class"
+    INTERFACE = "interface"
+
+
+# COM data source https://github.com/stevemk14ebr/COM-Code-Helper/tree/master
+VALID_COM_TYPES = {
+    ComType.CLASS: {"db_path": "assets/classes.json.gz", "prefix": "CLSID_"},
+    ComType.INTERFACE: {"db_path": "assets/interfaces.json.gz", "prefix": "IID_"},
+}
+
+com_db_cache: Dict[ComType, Dict[str, List[str]]] = {}
+
+
+def load_com_database(com_type: ComType) -> Dict[str, List[str]]:
+    com_db_path: Path = capa.main.get_default_root() / VALID_COM_TYPES[com_type]["db_path"]
+
+    if com_type in com_db_cache:
+        # If the com database is already in the cache, return it
+        return com_db_cache[com_type]
```

Comment:
could use `@functools.cache` decorator on this function and not have to implement the cache yourself.

Reply:
actually, please use `@functools.lru_cache` over `@functools.cache` because that one was added in 3.9.